wip-trampoline.md

```diff
@@ -0,0 +1,138 @@
+# Trampoline Onion Routing
+
+This document describes the additions we make to the existing onion routing
+to enable the use of trampoline hops.
+
+## Packet Structure
+
+### Trampoline Onion
+
+We define a fixed-size trampoline onion packet that can be nested inside onion
+packets. We define `N` to be the maximum number of trampoline nodes (excluding
+the sender node and the destination node).
+
+The trampoline packet always contains `2*N*FRAME_SIZE + 66` bytes.
```

Comment:
I'm not clear where the `2*FRAME_SIZE` per hop comes from.

Reply:
I see, it comes from the requirement of 45 bytes of payload, however if we just describe this as a TLV payload the rest resolves itself. That is if we just say we have `N` frames available, and allow any trampoline hop to use as many as they want we're not imposing this limitation right away.

Who knows, we might come up with a more concise way of addressing a node, and the amount to forward and we'd suddenly fit into a single frame.

Long story short: I think it's a mistake to hard-code the `2*FRAME_SIZE` here.

Reply:
This was an old commit and was removed afterwards.
Good news is that with yesterday's proposal during the IRC meeting, we're getting rid of `FRAME_SIZE` entirely ;)